python/pyspark/ml/param/_shared_params_code_gen.py

```diff
@@ -58,6 +61,9 @@ def __init__(self):
     if defaultValueStr is not None:
         template += '''
         self._setDefault($name=$defaultValueStr)'''
+        doc += '''(Default $friendlyDefault)'''
+        if friendlyDefault is None:
+            friendlyDefault = defaultValueStr
```

Comment:
Except for ```seed```, other params have the same ```defaultValueStr``` and ```friendlyDefault```. I don't think it's necessary to use two variables because ```defaultValueStr``` is already string type and can be used in the doc.
For param ```seed```, I think ```hash(type(self).__name__)``` is more clear than ```hash of type name``` for users to understand the default value of this param.